### PR TITLE
feat(lb): add data source for lb

### PR DESCRIPTION
**Pull request by remyleone**
It would be useful to have #752 so that we can generate the documentation for all resources easily.

### CODE REVIEW

File: scaleway/data_source_lb.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+func dataSourceScalewayLbRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := lbAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	lbID, ok := d.GetOk("lb_id")
+	if !ok { // Get LB by region.
```
Comment (jerome-quere) [inline scaleway/data_source_lb.go:42]:
```suggestion
	if !ok { // Get LB by name.
```

File: scaleway/data_source_lb.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		res, err := api.ListLBs(&lb.ListLBsRequest{
+			Region:    region,
+			Name:      expandStringPtr(d.Get("name")),
+			ProjectID: expandStringPtr(d.Get("project_id")),
+		}, scw.WithContext(ctx))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if len(res.LBs) == 0 {
+			return diag.FromErr(fmt.Errorf("no lbs found with the address %s", d.Get("ip_address")))
```
Comment (jerome-quere) [inline scaleway/data_source_lb.go:52]:
```suggestion
			return diag.FromErr(fmt.Errorf("no lbs found with the name %s", d.Get("name")))
```

File: docs/data-sources/lb.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+- `id` - The ID of the load-balancer.
+- `ip_address` -  The load-balance public IP Address
+- `organization_id` - The organization ID the load-balancer is associated with.
+- `tags` - (Optional) The tags associated with the load-balancers.
+- `type` - (Required) The type of the load-balancer.
```
Comment (jerome-quere) [inline docs/data-sources/lb.md:74]: Wrong copy/paste ?

File: docs/data-sources/lb.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+```
+
+## Argument Reference
+
+- `ip_address` - (Optional) The IP address.
+  Only one of `ip_address` and `lb_id` should be specified.
+
+- `lb_id` - (Optional) The ID.
+  Only one of `name` and `lb_id` should be specified.
```
Comment (QuentinBrosse) [inline docs/data-sources/lb.md:31]:
```suggestion
  Only one of `ip_address` and `lb_id` should be specified.
```

File: docs/data-sources/lb.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+# Get info by ID
+data "scaleway_lb" "by_id" {
+  lb_id = "11111111-1111-1111-1111-111111111111"
+}
+```
+
+## Argument Reference
+
+- `ip_address` - (Optional) The IP address.
```
Comment (jerome-quere) [inline docs/data-sources/lb.md:27]: Should be name ?

**Review by jerome-quere — CHANGES_REQUESTED** [review]

File: docs/data-sources/lb.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+- `project_id` - (Defaults to [provider](../index.md#project_id) `project_id`) The ID of the project the LB IP is associated with.
+
+## Attributes Reference
+
+In addition to all above arguments, the following attributes are exported:
+
+- `reverse` - The reverse domain associated with this IP.
+
+- `lb_id` - The associated load-balance ID if any
```
Comment (QuentinBrosse) [inline docs/data-sources/lb.md:43]: Already in `Argument Reference` section

**Review by QuentinBrosse — CHANGES_REQUESTED** [review]

**Review by QuentinBrosse — CHANGES_REQUESTED** [review]

File: scaleway/data_source_lb.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	return &schema.Resource{
+		ReadContext: dataSourceScalewayLbRead,
+		Schema:      dsSchema,
+	}
+}
+
+func dataSourceScalewayLbRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := lbAPIWithRegion(d, meta)
```
Comment (Monitob) [inline scaleway/data_source_lb.go:36]: lb do not support region anymore since version 2.1.0. I'm gonna update this branch.

**Review by Monitob — APPROVED** [review]

**Pull request merged**